### PR TITLE
show HTTP method for an endpoint

### PR DESCRIPTION
**Pull request by mszczygiel**
Hi, I think it's convenient method to have. My use case is for creating spans, but I think there may be other as well.

### CODE REVIEW

**Comment by mszczygiel** [comment]
sorry. It does not make sense for multiple supported methods.

**Comment by adamw** [comment]
it's not possible to have multiple supported methods :) the `.in` are a conjunction, not alternatives. So specifying both `.get` and `.post` would be an error.

File: core/src/main/scala/sttp/tapir/Endpoint.scala [hunk 1]
```diff
@@ -127,6 +127,16 @@ case class Endpoint[I, E, O, +S](input: EndpointInput[I], errorOutput: EndpointO
     */
   def showRaw: String = toString
 
+  def showMethod: String = {
+    def findMethod(input: EndpointInput[_]): Option[Method] = input match {
+      case FixedMethod(m)   => Some(m)
+      case Multiple(inputs) => inputs.flatMap(findMethod).headOption
```
Comment (adamw) [inline core/src/main/scala/sttp/tapir/Endpoint.scala]: I think we would also need to support mapped inputs?

**Comment by mszczygiel** [comment]
> it's not possible to have multiple supported methods :) the `.in` are a conjunction, not alternatives. So specifying both `.get` and `.post` would be an error.

I wasn't aware of that (totally new to tapir). I was misled, because I could do something like `endpoint.post.get.put` and this has details like `Endpoint(in: POST GET PUT, errout: -, out: -)`. What would you suggest returning from `showMethod` in such case?

I briefly checked and for `AkkaHttp` just 404 is returned when calling such endpoints.

**Comment by adamw** [comment]
It's a good question. One option is to fail with an exception at run-time, as this is an invalid mapping, and the error can't be caught at compile-time. But that's a separate issue :)

**Comment by adamw** [comment]
I think returning an `Option` is a good choice for this method

**Comment by mszczygiel** [comment]
OK, I think it's a good idea. I've created an issue https://github.com/softwaremill/tapir/issues/433 and I think I will be able to contribute it sometime later this week.

So maybe for now it's a good idea for `showMethod` to return any method present in a chain (which current implementation does)?

**Comment by adamw** [comment]
Thanks! Released as 0.12.21

**Pull request merged**